src/posts/2020-08-20-an-indepth-guide-to-aria-roles.md

```diff
@@ -0,0 +1,190 @@
+<button>Show Tooltip</button>  
+<div role="tooltip">Notifications</div>  
+```
+
+- View the [full list of available Document Structure roles](https://www.w3.org/TR/wai-aria/#document_structure_roles).
+
+### Landmark Roles
+
+Landmark roles identify content areas within a page, and as the name implies serve as navigational landmarks for the types of content areas that users may be interested in. These content areas include the page's primary content, collections of navigation links, and search functions to name a few.
+
+Most Landmark roles may also be reasonably associated with native HTML tags. For example:
+
+- The `banner` role can be associated with the `<header>` tag
+- The `complementary` role can be associated with the `<aside>` tag
+- The `contentinfo` role can be associated with the `<footer>` tag
+
+Because the support for some of the above native HTML elements may not be supported in older browsers or some screen readers, it is recommended that the role and native element be paired whenever possible.
+
+#### Navigation role example
+
+```html
+<nav role="navigation">
+  <ul>
+    <li><a href="#a">Dexter</a></li>
+    <li><a href="#b">Doctor Who</a></li>
+    <li><a href="#c">Futurama</a></li>
+  </ul>  
+</nav>
+```
+
+#### Form role example
+
+```html
+<form role="search">
+  <input type="text" name="search" />
+  <input type="submit" name="button" value="Search" />
+</form>
+```
+
+#### Footer role example
+
+```html
+<footer role="contentinfo">
+  <p>&copy; 2020 Small Business Ltd. All rights reserved.</p>
+</footer>
+```
+
+- View the [full list of available Landmark roles](https://www.w3.org/TR/wai-aria/#landmark_roles).
+
+### Live Region Roles
+
+Live Region roles are used to alert screen readers and assistive technologies of dynamic changes to the contents of a page so that they can be announced to users. The reason for this is that dynamic changes may be visually noticeable, but that may not be the case for screen reader users for .
+
+#### Alert role example
+
+The most commonly used Live Region role is the `alert` role.
+
+```html
+<div role="alert">
+  <h3>Error</h3>
+  <p>Autosave failed.</p>
+</div>
+```
+
+When using the `alert` role, keep in mind that it should not be a focusable element and it should not require users to close it.
+
+- View the [full list of available Live Region roles](https://www.w3.org/TR/wai-aria/#live_region_roles).
+
+### Widget Roles
+
+Widget roles are used to describe common interactive patterns or those that currently lack semantic equivalents in HTML. In most cases, these interactive patterns will be Javascript-driven. Widget roles should not be confused with `role="widget"` which is itself an abstract role.
+
+#### Tabs role example
+
+A common interactive pattern that relies on Widget roles is the very familiar Tabs pattern. Tabs rely on three Widget roles, namely `tab`, `tabpanel`, and `tablist`.
+
+```html
+<div>
+  <div role="tablist">
+    <button role="tab">Apples</button>
```

Comment:
again, while understood the post is about roles specifically,  code examples should indicate other necessary attributes for such patterns to work properly. Or, there should be an explicit call out made for people to learn more about other necessary attributes

Reply:
Updated the example